### PR TITLE
Feature: Reusable Pin Input View

### PR DESCRIPTION
**Pull request by fradenza**
### Breaking changes:
- Now `getCell()` method in `NestedTableViewConnector` allow to return `nil`.
- Add `tintColor` in `TextFieldStyle`, you can add the code bellow to fix your TextField style:
```
open var tintColor = UIColor(red: 0.0, green: 122.0/255.0, blue: 1.0, alpha: 1.0)
```

### Changes that need to be verified:
- Fix last status in form validation, need to verify in current project. Please do smoke test in your some form validations and make sure there is no layout and functionality defect.
- Add new `CoreConfig`, called `isAutoRenderOnEveryViewWillAppear` as an identifier to calling `render` method in every `viewWillAppear`. Other than that, `viewWillAppear` will be called when app goes to foreground. The default is `false`, so the current project does not need to add this config if you don't want to this changes reflect to your current app behaviour. Please do smoke test to run through between your screen and make sure there is no layout defect and functionality defect.

### Others changes:
- Create reusable `PinInputView` to handle input like OTP or PIN.
- Fix `TimeCounter` when app in background.
- Create String extension to accessing string by index.

### CODE REVIEW

File: OLCore/Classes/Views/PinInput/PinPanView.swift [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        addSubview(bottomLineView)
+    }
+
+    public func getValue() -> String {
+        return value
+    }
+
+    public func setValue(_ value: String) {
+        if self.value != DefaultValue.emptyString { return }
+        if value.count != 1 { return }
```
Comment (aldoonelabs) [inline OLCore/Classes/Views/PinInput/PinPanView.swift]: den ini max digit perlu di constant nggak?

File: OLCore/Classes/Views/PinInput/PinInputView.swift [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            addSubview(panView)
+        }
+    }
+
+    private func renderTextField() {
+        textField.style = InvisibleTextFieldStyle()
+        textField.delegate = self
+        textField.maxLength = length
+        textField.keyboardType = keyboardType
+        textField.addTarget(self, action: #selector(PinInputView.didChangedValue), for: .editingChanged)
```
Comment (aldoonelabs) [inline OLCore/Classes/Views/PinInput/PinInputView.swift]: ini dienter aja, udah lebih dari 2 den

File: OLCore/Classes/ViewControllers/ViewController/ViewController.swift [hunk 3]
```diff
@@ -52,6 +56,28 @@ open class ViewController: UIViewController {
...
+        }
+    }
+
+    private func removeForegroundObserver() {
+        guard let foregroundObserver = foregroundObserver else { return }
+        NotificationCenter.default.removeObserver(foregroundObserver)
+    }
+
+    private func setupForegroundObserver() {
+        foregroundObserver = NotificationCenter.default.addObserver(forName: UIApplication.willEnterForegroundNotification, object: nil, queue: .main) { [unowned self] notification in
```
Comment (aldoonelabs) [inline OLCore/Classes/ViewControllers/ViewController/ViewController.swift]: parameter di addObserver di enter aja

File: OLCore/Classes/Views/PinInput/PinPanView.swift (hunk 1 above)
Comment (fradenza) [reply OLCore/Classes/Views/PinInput/PinPanView.swift]: maksudnya angka `1` ya? ngga perlu kalo ini karna Pan Input udah pasti cuma 1.

**Pull request merged**